src/ragas/experimental/testset/transforms/splitters/headline.py

```diff
@@ -0,0 +1,52 @@
+import typing as t
+from dataclasses import dataclass
+
+from ragas.experimental.testset.graph import Node, Relationship
+from ragas.experimental.testset.transforms.base import Splitter
+
+
+@dataclass
+class HeadlineSplitter(Splitter):
```

Comment:
This I see handles single-level headline splitting, ideally, we might need an iterative one but let's leave that for now. Also, we would need normal hierarchical splitters like ones in llama-index or langchain.

Reply:
I've noted these down these will have to address them is separate PRs